packages/js/tracks/src/test/index.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Internal dependencies
+ */
+import { recordEvent } from '..';
+import { validateEventNameAndProperties } from '../utils';
+
+const eventName = 'my_event_name';
+const props = {
+	test: 'test value',
+};
+
+describe( 'recordEvent', () => {
+	let windowSpy: jest.SpyInstance;
+	const recordEventMock = jest.fn();
+
+	beforeEach( () => {
+		windowSpy = jest.spyOn( window, 'window', 'get' );
+		windowSpy.mockImplementation( () => ( {
+			wcTracks: {
+				recordEvent: recordEventMock,
+			},
+		} ) );
+	} );
+
+	afterEach( () => {
+		windowSpy.mockRestore();
+	} );
+
+	it( 'should record an event without props', () => {
+		recordEvent( eventName, {} );
+		expect( recordEventMock ).toHaveBeenCalledWith( eventName, {} );
+	} );
+
+	it( 'should record an event with props', () => {
+		recordEvent( eventName, props );
+		expect( recordEventMock ).toHaveBeenCalledWith( eventName, props );
+	} );
+
+	it( 'should not pass the validation an event without name', () => {
+		const response = validateEventNameAndProperties( '', props );
+		expect( response ).toBe( false );
+	} );
+
+	it( 'should not pass the validation an event with an invalid name', () => {
+		const nameWithDash = validateEventNameAndProperties(
+			'event-name',
+			props
+		);
+		const nameWithSpace = validateEventNameAndProperties(
```

Comment:
Nitpicky and I know it's a bit more verbose, but I tend to favor separating these out into individual tests.  A little easier to read and debug when an error does occur.

Reply:
I finally removed these tests since the validation now [is not accessible](https://github.com/woocommerce/woocommerce/pull/34005/files#r934606856) from here.